specification/app/resource-manager/Microsoft.App/preview/2024-08-02-preview/examples/JavaComponents_List.json

```diff
@@ -48,10 +48,40 @@
                 "fqdn": "myjavacomponent.myenvironment.test.net"
               }
             }
+          },
+          {
+            "id": "/subscriptions/8efdecc5-919e-44eb-b179-915dca89ebf9/resourceGroups/examplerg/providers/Microsoft.App/managedEnvironments/myenvironment/javaComponents/reddog",
+            "name": "reddog",
+            "type": "Microsoft.App/managedEnvironments/javaComponents",
+            "properties": {
+              "componentType": "SpringCloudGateway",
+              "provisioningState": "Succeeded",
+              "configurations": [
+                {
+                  "propertyName": "spring.cloud.gateway.enabled",
+                  "value": "true"
+                }
+              ],
+              "springCloudGatewayRoutes": [
+                {
+                  "id": "route1",
+                  "uri": "https://httpbin.org",
+                  "predicates": [
+                    "Path=/v1/{path}",
+                    "After=2024-01-20T17:42:47.789-07:00[America/Denver]"
```

Comment:
1. if these are known key types (Paths, After etc), can they be schematized into an object model?

2. Is there validation on your side on these values ?

3. Are these user input-ed values can you do not have any control over these values ?

Reply:
1 & 2, there are lots of key types. and yes we prefer to do validation in control plane and do not set any schema model in swagger. 

3. Yes it's customer's choice what predicates they want to set for their "gateway" component

Reply:
And seems that you missed my question about permission to edit labels, most of my colleagues are facing the same permission issue now. 
![image](https://github.com/Azure/azure-rest-api-specs/assets/15007516/f84ec2a9-43b9-49eb-9614-a725abed544c)

Reply:
> 1 & 2, there are lots of key types. and yes we prefer to do some minor validation in control plane which rollouts changes much faster and do not set any schema limit in swagger.
> 
> 3. Yes it's most likely a customer's choice what predicates they want to set for their "gateway" component, they basically can provide everything they want here, even if if can break the component itself.

I feel this is similar to AdditionalProperties.
"Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations. Please make this an array or provide an explanation for why you need this. "

Please provide an explanation if this cannot be schematized in anyway. To discuss further, please schedule a ARM API review office hours that happens every day. [aka.ms/](https://eng.ms/docs/products/arm/introduction/office_hours)

Reply:
For access, please check https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1716344171543?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1716344171543&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1716344171543

Reply:
"As part of the broader effort to tighten up security, we have made improvements to our permissions model that might have caused you to lose access.  Don't worry, it's easy to get back.  Simply join
[Azure SDK Partners](https://aka.ms/azsdk/join/azuresdkpartners)
and have your manager approve the request.  Within about a half hour, that SG will sync with
[azure-sdk-partners](https://github.com/orgs/Azure/teams/azure-sdk-partners)
on GitHub and you should have access to the repo again.  More details here:
[aka.ms/azsdk/access](https://aka.ms/azsdk/access)"

Reply:
Hey @ramoka178. Thx for the permission part. 

As for the review itself: 

> I feel this is similar to AdditionalProperties.

>"Use of additionalProperties is not allowed for properties owned by the service. The only time its ok to use it is when >the properties are pass thru (user defined) and not subject to any validations. Please make this an array or provide an >explanation for why you need this. "

You are right, these two properties "Filters" & "Predicates" are exactly similar as AdditionalProperties. And in our case, these two perperties exactly falls into the exceptional case you mentioned **"The only time its ok to use it is when the properties are pass thru (user defined) and not subject to any validations."**. They are user defined & pass thru & no subject to any validations. 

Not sure if this is reasonable for you.